src/libaktualizr/uptane/imagesrepository.cc

```diff
@@ -102,12 +124,36 @@ bool ImagesRepository::verifyTargets(const std::string& targets_raw) {
 }
 
 std::unique_ptr<Uptane::Target> ImagesRepository::getTarget(const Uptane::Target& director_target) {
-  auto it = std::find(targets.targets.cbegin(), targets.targets.cend(), director_target);
-  if (it == targets.targets.cend()) {
-    return std::unique_ptr<Uptane::Target>(nullptr);
-  } else {
+  // Search for the target in the top-level targets metadata.
```

Comment:
It might be nice (but I can try after this is merged) to turn this logic into an iterator. Then I think the logic of this function would become a std::find call to the new iterator

Reply:
Sounds feasible. But just so you know, I'm expecting this particular function to get reworked again when we actually get around to supporting nested delegations. It may also get more complex if/when we support prioritization and/or [TAP 3](https://github.com/theupdateframework/taps/blob/master/tap3.md) (multi-role delegations).